### PR TITLE
[IMP] move some notions from agreement_legal to agreement

### LINKED ISSUE #382: move some concepts out of agreement_legal
For a development I'm working on, I need some concepts on agreements that are introduced by agreement_legal, but without all the bells and whistles that are provided by that module. 

mainly:

* the agreement.type model (without necessary the subtypes)
* the is_template field on agreement

I propose making a PR which would move these definitions in the base agreement module on which agreement_legal depends. Alternatively, I could create a new module, but it seems a pity to tackle adding a dependency in agreement_legal.

### PR DESCRIPTION
**Pull request by gurneyalex**
we move the `is_template` field definition and the `agreement.type` model from
the `agreement_legal` module to the `agreement` module.

The fields are not displayed by default, unless the feature is enabled through a
technical feature group, this is configurable in the `agreement_sale` module (because agreement
in itself has no UI, and `agreement_legal` enables the feature by default)

Closes: #382

### CODE REVIEW

**Review by max3903 — APPROVED** [review]
LGTM. No Test

**Comment by bealdav** [comment]
No security rule for agreement.type when agreement_legal not installed.

Then impossible to complete data

Screenshot from 2019-09-30 09-47-10

pending evaluation ...

File: agreement_legal/views/agreement_type.xml
Comment (bealdav) [inline agreement_legal/views/agreement_type.xml:29]: @gurneyalex Partially off-topic of your PR but there is a missing part in agreement_sale as documented in the readme "This module adds Agreement field to Sales Orders"
Please could you move this view
https://github.com/OCA/contract/blob/12.0/agreement_legal_sale/views/sale_order.xml
to agreement_sale
But please add a group to agreement_template_id field please.
Thanks

**Review by bealdav — CHANGES_REQUESTED** [review]
Ok on my side, except the 2 points explained.

Nice work. 

With these modules agreement and agreement_sale now serves at least 3 uses cases (c2c, opensource integrators and akretion)
OCA proves one more time they can managed modularity across several integrators.

Thanks a lot

**Comment by bealdav** [comment]
Please @alexis-via could you have a look here for an approval, please

**Comment by bealdav** [comment]
Other element found 

is_template field is visible

agree

Whatever settings on it. Is it the behavior you wanted to implement ?

tmp

**Comment by gurneyalex** [comment]
thanks for the review @bealdav I'll take care ot this tomorrow.

**Comment by bealdav** [comment]
second review @gurneyalex, element https://github.com/OCA/contract/pull/383#issuecomment-536458890 not solved but maybe it's intentional ?

**Comment by gurneyalex** [comment]
@bealdav All your comments have been taken into account. If this is ok with you we can probably merge and move on :)

**Review by bealdav — APPROVED** [review]
Runbot is broken, but ok for me
Thanks a lot

File: agreement/models/agreement_type.py
Comment (i-vyshnevska) [inline agreement/models/agreement_type.py:8]: @gurneyalex  you need an access rule for type

**Pull request merged**
